app/src/main/java/com/sdp/movemeet/view/main/MainActivity.java

```diff
@@ -29,29 +29,21 @@
     @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE)
     public static boolean enableNav = true;
 
-    TextView fullName, email, phone;
     FirebaseAuth fAuth;
-    FirebaseFirestore fStore;
-    String userId;
-
-    DrawerLayout drawerLayout;
-    NavigationView navigationView;
-    Toolbar toolbar;
-    TextView textView;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        if (enableNav) new Navigation(this, R.id.nav_home).createDrawer();
-
         //The aim is to block any direct access to this page if the user is not logged
         fAuth = FirebaseAuth.getInstance();
         if (fAuth.getCurrentUser() == null) {
             startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
             finish();
         }
 
+        if (enableNav) new Navigation(this, R.id.nav_home).createDrawer();
```

Comment:
Does it change something to have the creation of the Navigation drawer after the user check ?

Reply:
Yes, it really changes because the navigation will try to retrieve information from a user. But if this one is null, there will be an error when creating the navigation. By reversing the order, we check that a user is connected and therefore that we have an authentication instance to create the navigation using this instance.